### PR TITLE
 Support PyPy

### PR DESCRIPTION
**Pull request by yuvalbar**
This PR allows PyMSSQL to compile on PyPy. It is based on this PR: https://github.com/pymssql/pymssql/pull/309. I rebased the branch on the current master and made the changes requested in the linked PR to ensure it compiles properly.

### CODE REVIEW

**Comment by yuvalbar** [comment]
Tests seem to be failing since this is a PR from a fork, but I validated it builds successfully on my machine.

**Comment by yuvalbar** [comment]
@damoxc @ramiro Just pinging in case you missed this. What do we need to do to get this merged?

**Comment by rsyring** [comment]
@yuvalbar, please see: https://github.com/pymssql/pymssql/issues/516

**Comment by yuvalbar** [comment]
@rsyring I created a native branch with all the changes in this repo to check if the build passes. If it does, I'll merge it - unless you have any objections. https://travis-ci.org/pymssql/pymssql/jobs/301628701

File: src/pypy_helpers.h [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#include "Python.h"
+
+/*
+ PyByteArrayCheck is not present in PyPy 2.5.1
+*/
+#ifndef PyByteArray_Check
+#define PyByteArray_Check(self) (0)
+#endif
```
Comment (rsyring) [inline src/pypy_helpers.h]: Would like to see a newline at the end of each file.

**Comment by rsyring** [comment]
@yuvalbar could we update CI so that we are also doing a test run with PyPI?

**Comment by rsyring** [comment]
@yuvalbar Thanks.  Did the tests get updated to run these changes?  Do you just want to create a new PR with the native branch so we can see the tests results in the PR?

**Comment by yuvalbar** [comment]
Closed in favor of https://github.com/pymssql/pymssql/pull/517

**Pull request closed without merging**